pkg/controller/decorator/controller.go

```diff
@@ -349,6 +350,32 @@ func (c *decoratorController) enqueueParentObjectAfter(obj interface{}, delay ti
 
 func (c *decoratorController) updateParentObject(old, cur interface{}) {
 	// TODO(enisoc): Is there any way to avoid resyncing after our own updates?
+	if parentOld, ok := old.(*unstructured.Unstructured); ok {
+		parentKind := parentOld.GetKind()
+		parentAPIVersion := parentOld.GetAPIVersion()
+		for _, parent := range c.dc.Spec.Resources {
+			resource := c.resources.Get(parent.APIVersion, parent.Resource)
```

Comment:
```
resource := c.resources.Get(parent.APIVersion, parent.Resource)
```
you can do the API version from https://github.com/metacontroller/metacontroller/blob/master/pkg/apis/metacontroller/v1alpha1/types.go#L55 directly (conversion from GVK to GVR - or opposite - is needed but we do it in other places)